dces-drc-integration/src/test/java/uk/gov/justice/laa/crime/dces/integration/utils/DateConvertorTest.java

```diff
@@ -0,0 +1,28 @@
+package uk.gov.justice.laa.crime.dces.integration.utils;
+
+import org.junit.jupiter.api.Test;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class DateConvertorTest {
+
+    @Test
+    void convertToXMLGregorianCalendarFromLocalDate() {
+        LocalDate date = LocalDate.of(2023, 10, 1);
+        XMLGregorianCalendar xmlGregorianCalendar = DateConvertor.convertToXMLGregorianCalendar(date);
+        assertNotNull(xmlGregorianCalendar);
+        assertEquals("2023-10-01", xmlGregorianCalendar.toString());
+    }
+
+    @Test
+    void convertToXMLGregorianCalendarWhenLocalDateIsNull() {
+        LocalDate date = null;
+        XMLGregorianCalendar xmlGregorianCalendar = DateConvertor.convertToXMLGregorianCalendar(date);
+        assertNull(xmlGregorianCalendar);
+    }
+}
```

Comment:
Do we need to add a test for InValid date? I noticed there is a try catch in the method, the catch is never tested, so at the moment, sonar is reporting 60% test coverage, I think by creating exception condition it could get over the magic 80% and probably hit 100.
If you excuse my copilot Java something like this
`   @Test
    public void convertToXMLGregorianCalendarWhenInvalidDate() {
        // This test is tricky because the method does not throw an exception for invalid dates
        // Instead, it logs an error and returns null. We can simulate this by mocking the DatatypeFactory.
        LocalDate date = LocalDate.of(2023, 10, 9);
        try {
            DatatypeFactory factory = DatatypeFactory.newInstance();
            XMLGregorianCalendar xmlDate = factory.newXMLGregorianCalendar(date.toString());
            assertNotNull(xmlDate, "Expected non-null XMLGregorianCalendar for valid date");
        } catch (Exception e) {
            fail("Exception should not be thrown for valid date");
        }
    }`